homeassistant/components/rachio/switch.py

```diff
@@ -276,3 +288,98 @@ def _handle_update(self, *args, **kwargs) -> None:
         """Unsubscribe from updates."""
         if self._undo_dispatcher:
             self._undo_dispatcher()
+
+
+class RachioSchedule(RachioSwitch):
+    """Representation of one fixed schedule on the Rachio Iro."""
+
+    def __init__(self, person, controller, data, current_schedule):
+        """Initialize a new Rachio Schedule."""
+        self._id = data[KEY_ID]
+        self._schedule_name = data[KEY_NAME]
+        self._duration = data[KEY_DURATION]
+        self._schedule_enabled = data[KEY_ENABLED]
+        self._summary = data[KEY_SUMMARY]
+        self._current_schedule = current_schedule
+        super().__init__(controller, poll=False)
+        self._state = self.schedule_id == self._current_schedule.get(KEY_SCHEDULE_ID)
+        self._undo_dispatcher = None
+
+    @property
+    def schedule_id(self) -> str:
+        """How the Rachio API refers to the schedule."""
+        return self._id
+
+    @property
+    def name(self) -> str:
+        """Return the friendly name of the schedule."""
+        return f"{self._schedule_name} Schedule"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique id by combining controller id and schedule."""
+        return f"{self._controller.controller_id}-schedule-{self.schedule_id}"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to display."""
+        return "mdi:water"
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return the optional state attributes."""
+        return {
+            ATTR_SCHEDULE_SUMMARY: self._summary,
+            ATTR_SCHEDULE_ENABLED: self.schedule_is_enabled,
+            ATTR_SCHEDULE_DURATION: self._duration / 60,
+        }
+
+    @property
+    def schedule_is_enabled(self) -> bool:
+        """Return whether the schedule is allowed to run."""
+        return self._schedule_enabled
+
+    def turn_on(self, **kwargs) -> None:
+        """Start this schedule."""
+
+        self._controller.rachio.schedulerule.start(self.schedule_id)
+        _LOGGER.debug(
+            "Schedule %s started on %s", self.name, self._controller.name,
+        )
+
+    def turn_off(self, **kwargs) -> None:
+        """Stop watering all zones."""
+        self._controller.stop_watering()
+
+    def _poll_update(self, data=None) -> bool:
+        """Poll the API to check whether the schedule is running."""
+        self._current_schedule = self._controller.current_schedule
+        return self.schedule_id == self._current_schedule.get(KEY_SCHEDULE_ID)
+
+    def _handle_update(self, *args, **kwargs) -> None:
```

Comment:
Does this need to run in the executor thread pool? If not, we should decorate it with `@callback` imported from core.py and prefix the name with `async_`.

Reply:
I'm planning to add another PR that allows all schedules to be used instead of just fixed schedules. I'll address these changes in that PR.